crates/secret-service/src/tests.rs

```diff
@@ -0,0 +1,142 @@
+use std::{
+    net::{Ipv4Addr, SocketAddr, SocketAddrV4},
+    sync::Arc,
+    time::Duration,
+};
+
+use bitcoin::key::Secp256k1;
+use musig2::secp256k1::Message;
+use rand::{thread_rng, Rng};
+use secret_service_client::SecretServiceClient;
+use secret_service_proto::v1::traits::{OperatorSigner, P2PSigner, SecretService};
+use secret_service_server::{
+    run_server,
+    rustls::{
+        self,
+        pki_types::{CertificateDer, PrivatePkcs8KeyDer, ServerName, UnixTime},
+        ClientConfig, ServerConfig,
+    },
+};
+
+use crate::disk::Service;
+
+#[tokio::test]
+async fn e2e() {
+    let server_addr: SocketAddr = SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), 20000).into();
+    let server_host = "localhost".to_string();
+
+    let cert = rcgen::generate_simple_self_signed(vec!["localhost".into()]).unwrap();
+    let key = PrivatePkcs8KeyDer::from(cert.key_pair.serialize_der());
+    let cert = cert.cert.into();
+    let server_tls_config = ServerConfig::builder()
+        .with_no_client_auth()
+        .with_single_cert(vec![cert], key.into())
+        .expect("valid config");
+    let config = secret_service_server::Config {
+        addr: server_addr,
+        tls_config: server_tls_config,
+        connection_limit: None,
+    };
+    let service = Service::new_with_seed([0u8; 32]);
+
+    tokio::spawn(async move {
+        run_server(config, service.into()).await.unwrap();
+    });
+
+    let client_tls = ClientConfig::builder()
+        .dangerous()
+        .with_custom_certificate_verifier(SkipServerVerification::new())
+        .with_no_client_auth();
+    let client_config = secret_service_client::Config {
+        server_addr,
+        server_hostname: server_host,
+        local_addr: None,
+        tls_config: client_tls,
+        timeout: Duration::from_secs(1),
+    };
+
+    let client = SecretServiceClient::new(client_config)
+        .await
+        .expect("good conn");
+
+    // operator signer
+    let op_signer = client.operator_signer();
+    let pubkey = op_signer.pubkey().await.expect("good response");
+
+    let handles = (0..1000)
+        .map(|_| {
+            let secp_ctx = Secp256k1::verification_only();
+            let op_signer = op_signer.clone();
+            tokio::spawn(async move {
+                let to_sign = thread_rng().gen();
+                let sig = op_signer.sign(&to_sign).await.expect("good response");
+                assert!(secp_ctx
+                    .verify_schnorr(&sig, &Message::from_digest(to_sign), &pubkey)
+                    .is_ok());
+            })
+        })
+        .collect::<Vec<_>>();
+    for handle in handles {
+        handle.await.unwrap();
+    }
+
+    // p2p signer
+    let p2p_signer = client.p2p_signer();
+    p2p_signer.secret_key().await.expect("good response");
+}
```

Comment:
Can we also test a stake preimage, a wots 160 and 256 secret keys, and a secret nonce?